### PR TITLE
compatibility to TYPO3 versions 9 and 10

### PR DESCRIPTION
**Pull request by DavidBruchmann**

### CODE REVIEW

**Comment by featdd** [comment]
Hi @DavidBruchmann,

thanks for your pull request.

Could you please edit your commit messages using the proper prefixes like "[TASK] ..."?
Also please remove the merge commit, local changes should be rebased on the current state.

I will test your changes after this and will merge them if everything is alright.

Greetings
Daniel

**Comment by DavidBruchmann** [comment]
Sure, I'm glad about every additional things I can do ;-)

**Comment by DavidBruchmann** [comment]
cant rebase for changing the title without problem.
Commit 0c7474b is making problems and I don't know how the deleted code came in again.
So please check what you can use of my code and range it up according to your own opinion.

**Comment by DavidBruchmann** [comment]
I know its a bit difficult as I took the fork of @sascha-seyfert as base, so I'm confused now
image

File: Configuration/TypoScript/setup.txt [hunk 1]
```diff
@@ -17,7 +17,7 @@ page.includeJSFooter.mindshapecookiehint = EXT:mindshape_cookie_hint/Resources/P
 page.9877 = USER
 page.9877 {
   userFunc = TYPO3\CMS\Extbase\Core\Bootstrap->run
-  pluginName = Main
+  pluginName = CookieHint
```
Comment (featdd) [inline Configuration/TypoScript/setup.txt:20]: Why did you change the plugin name or why was this necessary?

Comment (DavidBruchmann) [reply Configuration/TypoScript/setup.txt:20]: In the definitions for configurePlugin and registerPlugin have been some changes required and complaints in InstallTool never have been all reasonable for me. I changed things so long till it worked. So i general it could work if you replace "CookieHint" everywhere with "Main", but "CookieHint" is more descriptive anyway in my opinion.

**Comment by featdd** [comment]
Hi @DavidBruchmann,

I tested your changes, the controller part doesn't apply on the latest state and I encountered errors in another extension using the FQN of the controller in the plugin registration in v9.5
It is deprecated but still allowed to use only the name so it will work for both.

I will deny your pull request but the compatibility should be already achieved, but I can only tell for sure making some testing.

But thanks for your pull request anyway.

Greetings
Daniel

**Comment by DavidBruchmann** [comment]
Thanks for verifying everything and caring about compatibility ;-)

**Comment by featdd** [comment]
@DavidBruchmann ps: I added the compatibility, not much changes necessary see: 86c4a79175b90690ba2a4d436cbc87646330e914

It will be pushed to TER after a quality check

Greetings and fun with the extension
Daniel

**Comment by DavidBruchmann** [comment]
Daniel, yes I know ;-)

The most work was sorting out my PR and dealing with it.
I saw that the controller is already empty since some commits, and I tried
it like that too without any problems.
A small issue I still have, I will open another issue for that, concerns
one line css only.

Thanks a lot, it's much easier handling an official compatible release than
with temporary solutions, so it's an important work - even if it's done
fast.
